### PR TITLE
Fixing issues with ScriptSettingsManager reset when the configuration factory returned a cached configuration instance

### LINKED ISSUE #2526: ScriptSettingsManager Caching issue
Currently the implementation of ScriptSettingsManager is caching values, meaning for example during placeholder specialization when environment values are changing, we won't see them. The v1 implementation of GetSetting always called `Environment.GetEnvironmentVariable` so would see updated values. The v2 implementation is calling into `IConfiguration` which is caching things.

Repro: Enable [this test](https://github.com/Azure/azure-functions-host/blob/8cf2debe0f4896f4433d96b828b6586a06554293/test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs#L122) and try to run these lines of code that set the environment values.

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #2526

### CODE REVIEW

**Comment by fabiocav** [comment]
This is a tactical fix to address the issue described in #2526 until we get rid of `ScriptSettingsManager`

**Comment by fabiocav** [comment]
@mathewc can you take a look at this? This addresses the issue you've raised and should unblock your standby scenarios (as well as Bala's)

File: src/WebJobs.Script/Config/ScriptSettingsManager.cs [hunk 1]
```diff
@@ -153,6 +154,24 @@ public virtual void SetSetting(string settingKey, string settingValue)
             }
         }
 
+        private IConfiguration InitializeConfiguration(Lazy<IConfiguration> currentConfiguration)
+        {
+            var configuration = _configurationFactory();
+
+            // If the factory returned a cached instance
+            // that is the same as the instance we previously had,
+            // reload the configuration on that instance.
```
Comment (mathewc) [inline src/WebJobs.Script/Config/ScriptSettingsManager.cs]: Recommend adding some comments indicating why we're doing this - i.e. to enable us to pick up environment settings that change at runtime.

Comment (fabiocav) [reply src/WebJobs.Script/Config/ScriptSettingsManager.cs]: Thought about that, this really applies to all configuration, though. Not just app settings.

**Comment by fabiocav** [comment]
I re-enabled your tests and the issue seems to be resolved, but I did run into another, unrelated failure. Figured you'd be in a good position to look at it :)

**Pull request merged**